src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,30 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateStart = LocalDate.parse(dateFrom, formatter);
+        LocalDate dateEnd = LocalDate.parse(dateTo, formatter);
+        StringBuilder salaryReport = new StringBuilder();
+        salaryReport.append("Report for period ").append(dateFrom).append(" - ")
+                .append(dateTo).append(System.lineSeparator());
+        for (String name : names) {
+            int salaryTotal = 0;
+            for (String record : data) {
+                String[] partOfRecord = record.split(" ");
+                LocalDate workingDay = LocalDate.parse(partOfRecord[0], formatter);
```

Comment:
It's a good practice to handle potential exceptions when parsing dates. Consider wrapping `LocalDate.parse(partOfRecord[0], formatter)` in a try-catch block to handle `DateTimeParseException` in case the date format is incorrect.